backend/store/ChatStore.js

```diff
@@ -2,123 +2,174 @@ const Message = require("../models/Message");
 
 module.exports = class User {
     static storeNewMessageGroup = async (adventureId, messages, dateTime) => {
-        try {
-            var paginationResult = await this.getPrevPagination(adventureId, dateTime);
-            var messageGroup = { 
-                adventureId, 
-                pagination: dateTime, 
-                prevPagination: paginationResult.payload ? paginationResult.payload : null,
-                messages: [messages]
-            }
-            messageGroup = new Message(messageGroup);
-            var result = await messageGroup.save();
-
-            return {
+        var result = {
+            code: 500,
+            message: "Server error"
+        }
+        
+        var paginationResult = await this.getPrevPagination(adventureId, dateTime);
+        var messageGroup = { 
+            adventureId, 
+            pagination: dateTime, 
+            prevPagination: paginationResult.payload ? paginationResult.payload : null,
+            messages: [messages]
+        }
+        messageGroup = new Message(messageGroup);
+        var messageResult = await messageGroup.save();
+        if (messageResult) {
+            result = {
                 code: 200,
                 message: "Group created successfully",
-                payload: result
+                payload: messageResult
             };
+            
         }
-        catch (err) {
-            return {
-                code: 500,
-                message: err
+        else {
+            result = {
+                code: 400,
+                message: "Group creation unsuccessful"
             };
         }
+        
+        return result;
     };
 
     static storeExistingMessageGroup = async (adventureId, message, dateTime) => {
+        var result = {
+            code: 500,
+            message: "Server error"
+        }
+
         var paginationResult = await this.getPrevPagination(adventureId, dateTime);
-        var result = await Message.findOneAndUpdate( // update pagination and add message to object array
-                                { adventureId, pagination: paginationResult.payload },
-                                { $set: { pagination: dateTime },  $push: { messages: message } },
-                                { new: true }
-                            );
-        return {
-            code: 200,
-            message: "Group updated successfully",
-            payload: result
-        };
+        var paginationObj = { pagination: dateTime };
+        var messagesObj = { messages: message };
+        var messageResult = await Message.findOneAndUpdate( // update pagination and add message to object array
+                            { adventureId: adventureId, pagination: paginationResult.payload },
```

Comment:
Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/kalebhui/gruwup/pullRequest?prid=9818454)
```suggestion
                                    { adventureId, pagination: paginationResult.payload },
```